src/analysis/stack-lattice.h

```diff
@@ -0,0 +1,187 @@
+#ifndef wasm_analysis_stack_lattice_h
+#define wasm_analysis_stack_lattice_h
+
+#include <deque>
+#include <optional>
+
+#include "lattice.h"
+
+namespace wasm::analysis {
+
+// Models the WASM value stack
```

Comment:
```suggestion
// Models the Wasm value stack.
```

Reply:
Also, the two lines of comments below it now look a little too generic. I would expect to see something like this:
```
// A lattice to model the Wasm value stack.
//
// This is templated on a given lattice, and then represents stacks of elements of
// that one. These stacks can represent things on the Wasm value stack, such as
// ... For example, imagine that we want to optimize ... then we would use this
// on a lattice ... and then the optimization would work like ...
```

(I have no idea how to fill in those blanks, but that's the shape I'm looking for :smile: )

Reply:
OTOH, there's nothing in this lattice that ties it to the Wasm value stack specifically. That's just the first application we have in mind. I could also see this modeling exception handler stacks, for instance. I think it would be best to document the properties of this lattice (i.e. comparisons and joins being elementwise from the top of the stack rather than from the bottom of the stack) in more detail and use the connection to the Wasm value stack as an illustrative example.

Reply:
Added some more clarification.